cve_bin_tool/checkers/libxslt.py

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/python3
+
+"""
+CVE checker for libxslt
+https://www.cvedetails.com/product/14676/Xmlsoft-Libxslt.html?vendor_id=1962
+"""
+from . import Checker
+
+
+class LibxsltChecker(Checker):
+    CONTAINS_PATTERNS = []
+    FILENAME_PATTERNS = [r"libxslt\.so"]
+    VERSION_PATTERNS = [r"libxslt ([0-9]+\.[0-9]+\.[0-9]+)"]
```

Comment:
for your centos test, I'm not finding this string at all:

```console
[terri@cedar lib]$ strings libxslt.so.1 | grep -i libxslt
xsltLibxsltVersion
libxslt.so.1
 fake node libxslt
/usr/lib/libxslt-plugins
LIBXSLT_PLUGINS_PATH
LIBXSLT_PLUGINS_PATH is %s
libxslt:test element test worked
libxslt (SAXON 6.2 compatible)
libxslt.so.1.1.28.debug
```

Searching by version has that as the only candidate:

```console
[terri@cedar lib]$ strings * | grep -i 1.1.28
strings: Warning: 'libxslt-plugins' is a directory
libxslt.so.1.1.28.debug
libxslt.so.1.1.28.debug
```

So you probably want to use that string at least for centos.  You'll likely want to check what you find in, say, the ubuntu package because it's possible that string is specific to the way centos builds packages.

Reply:
It looks like there's also a series of previous versions related to libxml2 which might be related as a backup pattern, but you'd have to see if that would also match all libxml libraries

```
LIBXML2_1.1.0
LIBXML2_1.1.1
LIBXML2_1.1.2
LIBXML2_1.1.3
LIBXML2_1.1.5
LIBXML2_1.1.7
LIBXML2_1.1.9
LIBXML2_1.1.18
LIBXML2_1.1.20
LIBXML2_1.1.23
LIBXML2_1.1.24
LIBXML2_1.1.25
LIBXML2_1.1.26
LIBXML2_1.1.27
```

It might be a libxml thing, but it's possible the libxslt stuff always keeps versions in step with libxml, so that might be ok.  Worth doing some looking if you want to use those strings.